omegaconf/basecontainer.py

```diff
@@ -798,3 +815,27 @@ def _update_types(node: Node, ref_type: type, object_type: Optional[type]) -> No
         if new_ref_type is not Any:
             node._metadata.ref_type = new_ref_type
             node._metadata.optional = new_is_optional
+
+
+def _instantiate_structured_config_impl(
+    retdict: Dict[str, Any], object_type: Type[Any]
+) -> Any:
+    from ._utils import get_structured_config_data
+
+    object_type_field_names = get_structured_config_data(object_type).keys()
```

Comment:
get_structured_config_data is recursive and expensive.
looks like all you need is the keys. can you create a few helper functions that will just give you the keys?

something like:
```python
get_structured_config_field_names()
get_dataclass_field_names()
get_attr_class_field_names()
```

Reply:
Done in https://github.com/omry/omegaconf/pull/502/commits/fa81a4d4bc35342ebd2a76942ce68653d1f41078